libgit/autogit_manager.go

```diff
@@ -0,0 +1,404 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libgit
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/eapache/channels"
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/kbfs/libfs"
+	"github.com/keybase/kbfs/libkbfs"
+	billy "gopkg.in/src-d/go-billy.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+type resetReq struct {
+	srcTLF     *libkbfs.TlfHandle
+	srcRepo    string
+	branchName string
+	dstTLF     *libkbfs.TlfHandle
+	dstDir     string
+	doneCh     chan struct{}
+}
+
+func (r resetReq) dstPath() string {
```

Comment:
Is it planed to use `resetReq.dstPath()` for something other than keys in `resetsInQueue` and `resetsInProgress`? If it's not gonna be used as paths, would it be clearer to rename this into something like `resetReq.id()`?